econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -1136,10 +1176,116 @@ def _heterogeneity_tree_string(self, Xtest, feature_index, *,
             A (1-alpha)*100% confidence interval is displayed.
         """
 
-        intrp, feature_names, treatment_names, _ = self._tree(False, Xtest, feature_index,
-                                                              max_depth=max_depth,
-                                                              min_samples_leaf=min_samples_leaf,
-                                                              min_impurity_decrease=min_impurity_decrease,
-                                                              alpha=alpha)
-        return intrp.export_graphviz(feature_names=feature_names,
-                                     treatment_names=treatment_names)
+        intrp, feature_names, _, _ = self._tree(False, Xtest, feature_index,
+                                                max_depth=max_depth,
+                                                min_samples_leaf=min_samples_leaf,
+                                                min_impurity_decrease=min_impurity_decrease,
+                                                alpha=alpha)
+
+        def hetero_data(tree, node_id):
+            return {'effect': _sanitize(tree.value[node_id])}
+        return _tree_interpreter_to_dict(intrp, feature_names, hetero_data)
+
+    def individualized_policy(self, Xtest, feature_index, *, n_rows=None, treatment_costs=0, alpha=0.1):
```

Comment:
When I use this function all of the other features are somehow coming up as NaN when the input data frame has an index.
```python
inds = np.arange(1000)
np.random.shuffle(inds)
X = pd.DataFrame(np.random.normal(0, 1, size=(1000, 2)), columns=['A', 'B'], index=inds)
y = np.random.normal(0, 1, size=1000)
ca_test = CausalAnalysis(feature_inds=['A'], categorical=[])
ca_test.fit(X, y)
ind_policy = ca_test.individualized_policy(X[:50], feature_index='A')
ind_policy
```

Reply:
Both fixed.